### PR TITLE
Make InterpolatingAdjoint and BacksolveAdjoint mass matrix compatible

### PR DESCRIPTION
**Pull request by YingboMa**

### CODE REVIEW

File: src/local_sensitivity/adjoint_common.jl [hunk 1]
```diff
@@ -127,27 +127,23 @@ function generate_callbacks(sensefun, g, λ, t, callback, init_cb)
   affect! = let isq = (sensealg isa QuadratureAdjoint), λ=λ, t=t, y=y, cur_time=cur_time, idx=length(prob.u0), F=factorized_mass_matrix
     function (integrator)
       p, u = integrator.p, integrator.u
-      # Warning: alias here! Be care with λ
+      # Warning: alias here! Be careful with λ
       gᵤ = isq ? λ : @view(λ[1:idx])
       g(gᵤ,y,p,t[cur_time[]],cur_time[])
-      if isq
```
Comment (ChrisRackauckas) [inline src/local_sensitivity/adjoint_common.jl]: you don't need isq?

Comment (YingboMa) [reply src/local_sensitivity/adjoint_common.jl]: `isq` is checked in line 131.

**Comment by YingboMa** [comment]
CI failure is unrelated.

**Comment by ChrisRackauckas** [comment]
One of the CI's just messed up.

**Comment by Vaibhavdixit02** [comment]
Looks like this broke DiffEqFlux DAEs? 

https://travis-ci.org/github/SciML/DiffEqFlux.jl/builds/667581115#L872

What needs to change there? 
cc: @abhigupta768

**Comment by ChrisRackauckas** [comment]
Looks like it needs a fix and a test for out of place functions.

**Comment by abhigupta768** [comment]
>Looks like it needs a fix 
I think in https://github.com/SciML/DiffEqFlux.jl/blob/e620ea99053cd6a818cd9a0b8deeca8af26c5842/src/neural_de.jl#L298 `autojacvec=DiffEqSensitivity.ReverseDiffVJP(true)` is the reason for the failure? Should it be `InterpolatingAdjoint()` only now?

**Comment by ChrisRackauckas** [comment]
No, it should be using ReverseDiffVJPs here, but we should figure out why it's erroring. It's not really known. @YingboMa can you look into this really quickly by the end of the day?

**Comment by Vaibhavdixit02** [comment]
@YingboMa did you get a chance to look at this?

**Pull request merged**